numalogic/config/factory.py

```diff
@@ -24,24 +26,26 @@
 from numalogic.models.threshold import StdDevThreshold, StaticThreshold, SigmoidThreshold
 from numalogic.postprocess import TanhNorm, ExpMovingAverage
 from numalogic.preprocess import LogTransformer, StaticPowerTransformer, TanhScaler
+from numalogic.registry import MLflowRegistry
```

Comment:
When we use a factory pattern to instantiate desired types of objects, avoid importing all potential types by default.  Try to import during runtime when a library is actually needed.

This comment applies to all usage of factory patterns.